craft_platforms/rock/_build.py

```diff
@@ -34,6 +34,9 @@ def get_rock_build_plan(
     :param platforms: the platforms structure in ``rockcraft.yaml``
     :param build_base: the build base, if provided in ``rockcraft.yaml``.
     """
+    if base == "bare" and build_base is None:
+        raise _errors.NeedBuildBaseError(base=base)
```

Comment:
This should be in our general build planner.

Reply:
hang on, many applications don't accept `base: bare` but use the general build planner

Reply:
The general build planner is currently accepting `base: bare` though, right?